### PR TITLE
[#4000] improvement(client-python): Support simple auth for PyGVFS

### PR DESCRIPTION
**Pull request by xloya**
### What changes were proposed in this pull request?

Support simple auth for gravitino client in PyGVFS. The integration test depends on this PR: #3876 #3931 . When #3876 #3931 is merged, I will add integration tests and docs for this PR.

### Why are the changes needed?

Fix: #4000 

### How was this patch tested?

Add UTs and ITs.

### CODE REVIEW

File: clients/client-python/gravitino/filesystem/gvfs.py [hunk 1]
```diff
@@ -80,15 +83,45 @@ class GravitinoVirtualFileSystem(fsspec.AbstractFileSystem):
...
+        auth_type = (
+            GVFSConfig.DEFAULT_AUTH_TYPE
+            if options is None
+            else options.get(GVFSConfig.AUTH_TYPE, GVFSConfig.DEFAULT_AUTH_TYPE)
+        )
+        if auth_type == GVFSConfig.DEFAULT_AUTH_TYPE:
+            self._client = GravitinoClient(
+                uri=server_uri,
+                metalake_name=metalake_name,
+                check_version=False,
```
Comment (jerryshao) [inline clients/client-python/gravitino/filesystem/gvfs.py:107]: Why is it False here? I think we should set it to True.

File: clients/client-python/gravitino/filesystem/gvfs_config.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+class GVFSConfig:
+    CACHE_SIZE = "cache_size"
+    DEFAULT_CACHE_SIZE = 20
+
+    CACHE_EXPIRED_TIME = "cache_expired_time"
+    DEFAULT_CACHE_EXPIRED_TIME = 3600
+
+    AUTH_TYPE = "auth_type"
+    DEFAULT_AUTH_TYPE = "simple"
```
Comment (jerryshao) [inline clients/client-python/gravitino/filesystem/gvfs_config.py:15]: I think we should document these configs.

**Comment by jerryshao** [comment]
Unrelated to the PR here, I'm curious how do you support Kerberos HDFS for python gvfs, have you tried it?

**Comment by xloya** [comment]
> Unrelated to the PR here, I'm curious how do you support Kerberos HDFS for python gvfs, have you tried it?

When I tested pygvfs locally, I directly connected to the online HDFS cluster which enabled Kerberos authentication. There are two ways I know:
1. Add some configurations in the `core-site.xml` of the Hadoop environment:
```
<property>
<name>hadoop.security.authentication</name>
<value>kerberos</value>
</property>
<property>
<name>hadoop.client.kerberos.principal</name>
<value>xxx@HADOOP.COM</value>
</property>

<property>
<name>hadoop.client.keytab.file</name>
<value>/tmp/xxx.keytab</value>
</property>
```
2. Execute kinit in the test local and log in to Kerberos

File: clients/client-python/gravitino/filesystem/gvfs.py (hunk 1 above)
Comment (xloya) [reply clients/client-python/gravitino/filesystem/gvfs.py:107]: Yeah, I will remove this change, it's just for testing.

File: clients/client-python/gravitino/filesystem/gvfs_config.py (hunk 2 above)
Comment (xloya) [reply clients/client-python/gravitino/filesystem/gvfs_config.py:15]: Yeah, I think we can review and merge this PR #3931 firstly, then I can add the new config docs.

**Comment by xloya** [comment]
@jerryshao This PR is ready, please take a look, thanks.

**Review by jerryshao — APPROVED** [review]
LGTM, thanks @xloya for your contribution.

File: clients/client-python/gravitino/filesystem/gvfs_config.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+"""
+Copyright 2024 Datastrato Pvt Ltd.
+This software is licensed under the Apache License version 2.
+"""
```
Comment (jerryshao) [inline clients/client-python/gravitino/filesystem/gvfs_config.py:4]: I think the license header should be updated.

Comment (xloya) [reply clients/client-python/gravitino/filesystem/gvfs_config.py:18]: Done

**Review by jerryshao — APPROVED** [review]

**Pull request merged**